helper/testhelpers/testhelpers.go

```diff
@@ -480,70 +547,105 @@ func enableDrPrimary(t testing.T, tc *vault.TestCluster) {
 	WaitForReplicationStatus(t, c.Client, true, func(secret map[string]interface{}) bool {
 		return secret["mode"] != nil && secret["mode"] == "primary"
 	})
+	tc.Logger.Info("enabled dr primary")
 }
 
-func getDrToken(t testing.T, tc *vault.TestCluster, id string) string {
+func getDrToken(t testing.T, tc *vault.TestCluster, id, secondaryPublicKey string) string {
 	req := map[string]interface{}{
 		"id": id,
 	}
+	if secondaryPublicKey != "" {
+		req["secondary_public_key"] = secondaryPublicKey
+	}
 	secret, err := tc.Cores[0].Client.Logical().Write("sys/replication/dr/primary/secondary-token", req)
 	if err != nil {
 		t.Fatal(err)
 	}
 	return secret.WrapInfo.Token
 }
 
-func (r *ReplicatedTestClustersBuilder) enablePerformanceSecondary(t testing.T) {
-	c := r.clusters.PerfSecondaryCluster.Cores[0]
+func EnablePerformanceSecondary(t testing.T, perfToken string, pri, sec *vault.TestCluster, updatePrimary, skipPoisonPill bool) string {
 	postData := map[string]interface{}{
-		"token":   r.perfToken,
-		"ca_file": r.clusters.PerfPrimaryCluster.CACertPEMFile,
+		"token":   perfToken,
+		"ca_file": pri.CACertPEMFile,
 	}
-	if r.clusters.PerfPrimaryCluster.ClientAuthRequired {
-		p := r.clusters.PerfPrimaryCluster.Cores[0]
+	if pri.ClientAuthRequired {
+		p := pri.Cores[0]
 		postData["client_cert_pem"] = string(p.ServerCertPEM)
 		postData["client_key_pem"] = string(p.ServerKeyPEM)
 	}
-	_, err := c.Client.Logical().Write("sys/replication/performance/secondary/enable", postData)
+	path := "sys/replication/performance/secondary/enable"
+	if updatePrimary {
+		path = "sys/replication/performance/secondary/update-primary"
+	}
+	_, err := sec.Cores[0].Client.Logical().Write(path, postData)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	WaitForReplicationState(t, c.Core, consts.ReplicationPerformanceSecondary)
+	sec.Logger.Info("enabled perf secondary, waiting for its replication state")
+	WaitForReplicationState(t, sec.Cores[0].Core, consts.ReplicationPerformanceSecondary)
 
-	r.clusters.PerfSecondaryCluster.BarrierKeys = r.clusters.PerfPrimaryCluster.BarrierKeys
-
-	// We want to make sure we unseal all the nodes so we first need to wait
-	// until two of the nodes seal due to the poison pill being written
-	WaitForNCoresSealed(t, r.clusters.PerfSecondaryCluster, 2)
-	EnsureCoresUnsealed(t, r.clusters.PerfSecondaryCluster)
-	WaitForActiveNode(t, r.clusters.PerfSecondaryCluster)
+	var perfSecondaryRootToken string
```

Comment:
We may want to, before going any further, add a function here that waits for replication to fully sync between the two clusters. I've seen failures happen when we access the API before audit salts have replicated down. there is this helper that checks for matching merkle roots without using the API https://github.com/hashicorp/vault/blob/ba5fe5d4f85d863ded9160732f2d503889f13425/helper/testhelpers/testhelpers.go#L741

Reply:
Done.